src/app/src/components/FacilityListItemsConfirmationTableRow.jsx

```diff
@@ -86,24 +87,34 @@ function FacilityListItemsConfirmationTableRow({
     ];
 
     const [matches, setMatches] = useState([]);
+
+    const filterUniqueMatches = matchList => {
```

Comment:
What do you think about using the useMemo hook to cache the result of a calculation between re-renders?
And move it to hooks.

Reply:
I've checked your suggestion and it didn't work as expected for me, probably because of the usage of the local `matches` state here that should be updated each time the component receives new items. Lets use `useMemo` optimization a bit later after resolving other major tech debt issues.

Reply:
Ok